### PR TITLE
Handling for getitem with boolean in meta, and other improvements

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85807

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: test/test_meta.py [hunk 1]
```diff
@@ -217,7 +217,6 @@ def test_assert(cond, msg):
         test_assert(isinstance(meta_r, torch.Tensor), f"but real {i}th result is Tensor")
         test_assert(meta_r.dtype == r.dtype, f"but real dtype was {r.dtype}")
         test_assert(meta_r.shape == r.shape, f"but real shape was {r.shape}")
-        # NOTE: stride checking is currently disabled
```
Comment (albanD) [inline test/test_meta.py:220]: delete also the line below?

**Review by albanD — APPROVED** [review]
SGTM

File: test/test_meta.py (hunk 1 above)
Comment (ezyang) [reply test/test_meta.py:220]: The issue is still relevant as we aren't checking strides still in many cases

**Pull request closed without merging**